google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/test_helpers/env/AbstractTestEnv.java

```diff
@@ -75,6 +90,38 @@ public abstract BigtableTableAdminClient getTableAdminClientForInstance(String i
 
   public abstract String getInstanceId();
 
+  /** Try to guess the primary cluster id */
+  public synchronized String getPrimaryClusterId() {
+    if (primaryClusterId != null) {
+      return primaryClusterId;
+    }
+
+    List<Cluster> clusters = getInstanceAdminClient().listClusters(getInstanceId());
+    Map<String, Cluster> byId = new HashMap<>();
+    for (Cluster cluster : clusters) {
+      byId.put(cluster.getId(), cluster);
+    }
+
+    Cluster cluster = null;
+
+    cluster = byId.get(getInstanceId());
+    if (cluster == null) {
+      // gcloud-devel setup
+      cluster = byId.get(getInstanceId() + "-cluster");
+    }
+    if (cluster == null) {
```

Comment:
seems duplicated?

Reply:
not duplicated, just different fallbacks:
- check if a cluster is named same as instance
- check if a cluster named as instance suffixed with -clsuter
- check if a cluster named as isntance suffixed with -c0

Reply:
sorry. Didn't see the map bit